src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaWriter.cs

```diff
@@ -400,6 +401,31 @@ internal void WriteOperationParameterElementHeader(IEdmOperationParameter parame
             }
         }
 
+        internal void WriteOperationParameterEndElement(IEdmOperationParameter parameter)
+        {
+            IEdmOptionalParameter optionalParameter = parameter as IEdmOptionalParameter;
+            if (optionalParameter != null && !(optionalParameter.VocabularyAnnotations(this.model).Where(a => a.Term == CoreVocabularyModel.OptionalParameterTerm).Count() > 0))
+            {
+                string defaultValue = optionalParameter.DefaultValueString;
+                List<EdmPropertyConstructor> properties = new List<EdmPropertyConstructor>();
+                var optionalValue = new EdmRecordExpression(properties);

```

Comment:
This is the one use of var that seems OK but I agree, I'd love to see us get out of the var habit.

---
In reply to: [123061452](https://github.com/OData/odata.net/pull/828#discussion_r123061452) [](ancestors = 123061452)